test/navigation.spec.js

```diff
@@ -723,45 +722,47 @@ module.exports.describe = function({testRunner, expect, playwright, FFOX, CHROMI
       await waitPromise;
       expect(resolved).toBe(true);
     });
+    it('should work for cross-process navigations', async({page, server}) => {
+      await page.goto(server.EMPTY_PAGE);
+      const waitPromise = page.waitForNavigation({waitUntil: []});
+      const url = server.CROSS_PROCESS_PREFIX + '/empty.html';
+      const gotoPromise = page.goto(url);
+      const response = await waitPromise;
+      expect(response.url()).toBe(url);
+      expect(page.url()).toBe(url);
+      expect(await page.evaluate('document.location.href')).toBe(url);
+      await gotoPromise;
+    });
   });
 
   describe('Page.waitForLoadState', () => {
     it('should pick up ongoing navigation', async({page, server}) => {
       let response = null;
       server.setRoute('/one-style.css', (req, res) => response = res);
-      const navigationPromise = page.goto(server.PREFIX + '/one-style.html');
-      await server.waitForRequest('/one-style.css');
+      await Promise.all([
+        server.waitForRequest('/one-style.css'),
+        page.goto(server.PREFIX + '/one-style.html', {waitUntil: []}),
+      ]);
       const waitPromise = page.waitForLoadState();
       response.statusCode = 404;
       response.end('Not found');
       await waitPromise;
-      await navigationPromise;
     });
     it('should respect timeout', async({page, server}) => {
-      let response = null;
       server.setRoute('/one-style.css', (req, res) => response = res);
-      const navigationPromise = page.goto(server.PREFIX + '/one-style.html');
-      await server.waitForRequest('/one-style.css');
+      await page.goto(server.PREFIX + '/one-style.html', {waitUntil: []});
       const error = await page.waitForLoadState({ timeout: 1 }).catch(e => e);
```

Comment:
This test could fake if waitForLoadState ran before the document committed with page.goto.